backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ImageTransferUpdater.java

```diff
@@ -50,12 +51,18 @@ public ImageTransfer updateEntity(ImageTransfer updates, Guid commandId, boolean
                     entity.setId(updates.getId());
                 }
                 if (updates.getPhase() != null) {
-                    // TODO: Validate that phase change is valid. For now just log.
-                    log.info("Updating image transfer '{}' phase from '{}' to '{}'",
-                            commandId,
-                            entity.getPhase(),
-                            updates.getPhase());
-                    entity.setPhase(updates.getPhase());
+                    if (ImageTransferPhase.validateTransition(entity.getPhase(), updates.getPhase())) {
```

Comment:
Looks like you got this check wrong - if I understand the validator correctly, it return true if the transition is valid. but here treat true as invalid, which does not make sense.